src/mongocxx/test/search_index_view.cpp

```diff
@@ -250,9 +273,295 @@ TEST_CASE("atlas search indexes prose tests", "") {
+        auto model3 = search_index_model(name3, definition3.view(), type3);
+
+        REQUIRE(model3.type().value() == "vectorSearch");
+        REQUIRE(siv.create_one(model3) == "test-search-index-case7-vector");
+
+        assert_soon([&siv, &model3](void) -> bool {
+            auto cursor = siv.list();
+            return does_search_index_exist_on_cursor_with_type(
+                cursor, model3, "vectorSearch", false);
+        });
+
+        std::cout << "Prose Test Case 7: Driver can successfully handle search index types when "
+                     "creating indexes SUCCESS"
+                  << std::endl;
+    }
+
+    // Prose test case #8.
+    SECTION("Driver requires explicit type to create a vector search index") {
+        bsoncxx::oid id;
+        auto coll0 = db.create_collection(id.to_string());
+        auto siv = coll0.search_indexes();
+        //   {
+        //     name: 'test-search-index-case8-error',
+        //     definition: {
+        //       fields: [
+        //          {
+        //              type: 'vector',
+        //              path: 'plot_embedding',
+        //              numDimensions: 1536,
+        //              similarity: 'euclidean',
+        //          },
+        //       ]
+        //     }
+        //   }
+        auto name = "test-search-index-case8-error";
+        auto definition =
+            make_document(kvp("fields",
+                              make_array(make_document(kvp("type", "vector"),
+                                                       kvp("path", "plot_embedding"),
+                                                       kvp("numDimensions", 1536),
+                                                       kvp("similarity", "euclidean")))));
+        auto model = search_index_model(name, definition.view());
+
+        REQUIRE_THROWS_WITH(siv.create_one(model),
+                            "Attribute mappings missing.: generic server error");
+
+        std::cout << "Prose Test Case 8: Driver requires explicit type to create a vector search "
+                     "index SUCCESS"
                   << std::endl;
     }
 }
+
+TEST_CASE("atlas search indexes tests", "") {
+    instance::current();
+
+    auto uri_getenv = std::getenv("MONGODB_URI");
+    if (!uri_getenv) {
+        WARN("Skipping - Test requires the environment variable: MONGODB_URI");
+        return;
+    }
+
+    client mongodb_client{uri{uri_getenv}};
+
+    database db = mongodb_client["test"];
+
+    SECTION("get fields from model") {
+        // {
+        //   name: 'test-search-index',
+        //   type: "search",
+        //   definition : {
+        //     mappings : { dynamic: false }
+        //   }
+        // }
+        auto name1 = "test-search-index";
+        auto type1 = "search";
+        auto definition1 = make_document(kvp("mappings", make_document(kvp("dynamic", false))));
+        auto model1 = search_index_model(name1, definition1.view(), type1);
```

Comment:
```suggestion
        const auto name1 = "test-search-index";
        const auto type1 = "search";
        const auto definition1 =
            make_document(kvp("mappings", make_document(kvp("dynamic", false))));
        const auto model1 = search_index_model(name1, definition1.view(), type1);
```
Apply const wherever applicable.